File: OutOfOffice.Contracts/Models/Employee.cs
namespace OutOfOffice.Contracts.Models;

public class Employee
{
    public int Id { get; set; }
    public string FullName { get; set; }
    public int SubdivisionId { get; set; }
    public int PositionId { get; set; }
    public int StatusId { get; set; }
    public int PeoplePartnerId { get; set; }
    public decimal OutOfOfficeBalance { get; set; }

    public Employee PeoplePartner { get; set; }
    public Position Position { get; set; }
    public EmployeeStatus Status { get; set; }
    public Subdivision Subdivision { get; set; }
}

File: outofoffice.client/src/app/models/employee.ts
import { ProjectEmployee } from "./project"

export interface Employee {
  id: number
  fullName: string
  outOfOfficeBalance: number
  peoplePartner: PeoplePartner | null
  position: Position
  status: EmployeeStatus
  subdivision: Subdivision
  projectEmployees: ProjectEmployee[]
}

export interface PeoplePartner {
  id: number
  fullName: string
}

export interface Position {
  id: number
  name: string
}

export interface EmployeeStatus {
  id: number
  name: string
}

export interface Subdivision {
  id: number
  name: string
}

export interface EmployeePayload {
  fullName: string
  outOfOfficeBalance: number
  positionId: number
  statusId: number
  subdivisionId: number
  peoplePartnerId: number | null
}
